shopinvader_membership/services/membership.py

```diff
@@ -41,21 +46,41 @@ def search(self, **params):
         """
         return self._paginate_search(**params)
 
-    def subscribe(self, _id):
+    @restapi.method(
+        [(["/<int:_id>/subscribe"], "GET")],
+        input_param=restapi.CerberusValidator({}),
+        output_param=restapi.CerberusValidator("_validator_return_subscribe"),
+    )
+    def get_subscribe(self, _id):
         """
+        DEPRECATED: you should use `subscribe` with a POST.
         Subscribe to a membership product with logged user
         :param _id: id of product.product
         :return: dict with invoice_id
         """
+        _logger.warning("DEPRECATED: You should use `subscribe` with a POST")
+        return self.post_subscribe(**{"membership_product_id": _id})
+
+    @restapi.method(
+        [(["/subscribe"], "POST")],
+        input_param=restapi.CerberusValidator("_validator_subscribe"),
+        output_param=restapi.CerberusValidator("_validator_return_subscribe"),
+    )
+    def post_subscribe(self, **params):
+        """
+        Subscribe to a membership product with logged user
+        :return: dict with invoice_id
+        """
+        product_id = params.get("membership_product_id")
         if not self._is_logged_in():
             raise UserError(_("A user should be logged"))
         membership_product = self.env["product.product"].search(
-            [("id", "=", _id), ("membership", "=", True)]
+            [("id", "=", product_id), ("membership", "=", True)], limit=1
         )
         if not membership_product:
-            raise UserError(_("No membership product found with id %s") % _id)
+            raise UserError(_("No membership product found with id %s") % product_id)
         wizard = self.env["membership.invoice"].create(
-            {"product_id": _id, "member_price": membership_product.list_price}
+            {"product_id": product_id, "member_price": membership_product.list_price}
         )
```

Comment:
One last thing: I'd rather avoid using the wizard and call `res.partner.create_membership_invoice()` instead. The code will be easier to understand and the wizard will stay where it needs to stay (i.e. in the UI realm and not the business logic realm).